themes/_core/js/checkout.js

```diff
@@ -71,6 +72,16 @@ function handleCheckoutStepChange() {
   );
 }
 
+function handleSubmitButton() {
+  // prevents rage clicking on submit button and related issues
+  $(document).on('click', '.js-current-step button[type="submit"]', function () {
+    setTimeout( () => $(this).prop('disabled',true) , 0); //defer so the form still submits
```

Comment:
I don't see the reason to use a setTimeout here.
You can add the `disabled` property without defering.

Reply:
If I don't the form won't submit anymore. I already tried it without timeout.

Reply:
I'm worried about side-effects on themes that are based on Classic Theme 🤔 

Am I too paranoid ?

Reply:
> I'm worried about side-effects on themes that are based on Classic Theme 🤔
> 
> Am I too paranoid ?

I don't think this would affect Classic based or any third party themes. We just disable a button via a proprety, it's pretty basic. Do you have a use case in mind when this would interfeer?

Reply:
If someone already implement this feature 😉 it could try to disable it twice

Reply:
> If someone already implement this feature 😉 it could try to disable it twice

Yes, that's true but it will still work. I just tried it: added the same function in `calssic/assets/js/custom.js` and I have no error.

Reply:
🎉

Reply:
If you added a setTimeout to fix a weird behaviour, the fix must be done elsewhere :/

Reply:
> If you added a setTimeout to fix a weird behaviour, the fix must be done elsewhere :/

You're right, but I don't know if it really is that weird. I think it's normal that you cannot submit a form if you disable its submit button first. Using the setTimout(0) I'm just sending the execution of `$(this).prop('disabled',true)` to the end of the queue, so it works.

Reply:
Hi @PierreRambaud 
I've altered this PR to use bootstrap `disabled` class instead (that uses `pointer-events: none` to disable links and buttons). It has [pretty good browser compatibility](https://caniuse.com/#feat=pointer-events) and we get rid of any setTimeout. WDYT?

Reply:
> Hi @PierreRambaud
> I've altered this PR to use bootstrap `disabled` class instead (that uses `pointer-events: none` to disable links and buttons). It has [pretty good browser compatibility](https://caniuse.com/#feat=pointer-events) and we get rid of any setTimeout. WDYT?

On further testing it doesn't work so well in FF, IE or Opera. It just disables the button visually (this could discourage pepole to click it again), but it's still clickable, so this still is not a fix.

Reply:
Much better :+1: